ocp-identity-providers/ldap.yml

```diff
@@ -0,0 +1,55 @@
+---
+apiVersion: v1
+kind: Template
+message: |-
+  The cluster ldap identity provider has been created
+metadata:
+  name: ldap-idp-template
+objects:
+- apiVersion: config.openshift.io/v1
+  kind: OAuth
+  metadata:
+    name: cluster
+  spec:
+    identityProviders:
+    - name: ldapidp
+      mappingMethod: claim
+      type: LDAP
+      ldap:
+        attributes:
+          id:
+          - dn
```

Comment:
For these different attributes -- would it be better to make them configurable as well? I know these are fairly standard and the default -- but I know a lot of times some places will change what attributes they use for different identifiers.

Maybe a bit of an edge case, but just wanted to see what everyone elses thoughts were.

Reply:
I agree, ideally we'd like to make these configurable. The only way I know of (in an openshift template) is to make it a valid json string.
Something like
```
attributes: {"id": ["dn"], "email": ["mail"], "name": ["cn"], "preferredUsername": ["uid"]}
```

Reply:
I've updated the PR with the above as the default.